src/main/java/com/exasol/releaserobot/ReleaseInteractor.java

```diff
@@ -30,25 +30,26 @@ public ReleaseInteractor(final ValidateUseCase validateUseCase,
     @Override
     public List<Report> release(final UserInput userInput) {
         final List<Report> reports = new ArrayList<>();
-        final ValidationReport validationReport = this.validateUseCase.validate(userInput);
+        final Report validationReport = this.validateUseCase.validate(userInput);
         reports.add(validationReport);
         if (!validationReport.hasFailures()) {
             LOGGER.info(() -> "Release started.");
-            final ReleaseReport releaseReport = this.makeRelease(userInput.getPlatformNames());
+            final Report releaseReport = this.makeRelease(userInput.getPlatformNames());
             logResults(Goal.RELEASE, releaseReport);
             reports.add(releaseReport);
         }
         return reports;
     }
 
-    private ReleaseReport makeRelease(final Set<PlatformName> platformNames) {
-        final ReleaseReport releaseReport = new ReleaseReport();
+    private Report makeRelease(final Set<PlatformName> platformNames) {
+        final Report releaseReport = new ReportImpl(ReportImpl.ReportName.RELEASE);
```

Comment:
I wonder if this could be done with a static factory method (not the design pattern from GoF).

Something like ReportImpl.createNewReleaseReport();

Maybe for a next PR if it's really necessary.